src/App.jsx

```diff
@@ -14,70 +17,68 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+export const App = () => {
+  const [state, setState] = useState(`Jam`);
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+  return (
+    <main className="section container">
+      { state === ''
+        ? (
+          <h1 className="title is-flex is-align-items-center">
+            No goods selected
+          </h1>
+        ) : (
+          <h1 className="title is-flex is-align-items-center">
+            {`${state} is selected`}
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+              className="delete ml-3"
+              onClick={() => setState('')
+              }
+            />
+          </h1>
+        )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(goodItem => (
+            <tr
+              data-cy="Good"
+              className={cn('Good', {
+                'has-background-success-light': state === goodItem,
+              })}
+              key={goodItem}
             >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+              <td>
+                <button
+                  data-cy={state === goodItem ? 'RemoveButton' : 'AddButton'}
+                  type="button"
+                  className={cn(
+                    'button', {
+                      'is-info': state === goodItem,
+                    },
+                  )}
+                  onClick={() => {
+                    if (state !== goodItem) {
+                      setState(goodItem);
+                    } else {
+                      setState('');
+                    }
+                  }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)